src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Unix.cs

```diff
@@ -2,108 +2,1037 @@
+                    throw new WebSocketException(SR.net_webstatus_ConnectFailure);
+                }
+
+                // Read each response header. Be liberal in parsing the response header, treating
+                // everything to the left of the colon as the key and everything to the right as the value, trimming both.
+                // For each header, validate that we got the expected value.
+                bool foundUpgrade = false, foundConnection = false, foundSecWebSocketAccept = false;
+                string subprotocol = null;
+                string line;
+                while (!string.IsNullOrEmpty(line = await ReadLineNoBufferingAsync(stream, cancellationToken).ConfigureAwait(false)))
+                {
+                    int colonIndex = line.IndexOf(':');
+                    if (colonIndex == -1)
+                    {
+                        throw new WebSocketException(WebSocketError.HeaderError, SR.net_WebSockets_HeaderError_Generic);
+                    }
+
+                    string headerName = line.SubstringTrim(0, colonIndex);
+                    string headerValue = line.SubstringTrim(colonIndex + 1);
+
+                    // The Connection, Upgrade, and SecWebSocketAccept headers are required and with specific values.
+                    ValidateAndTrackHeader(HttpKnownHeaderNames.Connection, "Upgrade", headerName, headerValue, ref foundConnection);
+                    ValidateAndTrackHeader(HttpKnownHeaderNames.Upgrade, "websocket", headerName, headerValue, ref foundUpgrade);
+                    ValidateAndTrackHeader(HttpKnownHeaderNames.SecWebSocketAccept, expectedSecWebSocketAccept, headerName, headerValue, ref foundSecWebSocketAccept);
+
+                    // The SecWebSocketProtocol header is optional.  We should only get it with a non-empty value if we requested subprotocols,
+                    // and then it must only be one of the ones we requested.  If we got a subprotocol other than one we requested (or if we
+                    // already got one in a previous header), fail. Otherwise, track which one we got.
+                    if (string.Equals(HttpKnownHeaderNames.SecWebSocketProtocol, headerName, StringComparison.OrdinalIgnoreCase) &&
+                        !string.IsNullOrWhiteSpace(headerValue))
+                    {
+                        string newSubprotocol = options.RequestedSubProtocols.Find(requested => string.Equals(requested, headerValue, StringComparison.OrdinalIgnoreCase));
+                        if (newSubprotocol == null || subprotocol != null)
+                        {
+                            throw new WebSocketException(
+                                WebSocketError.UnsupportedProtocol,
+                                SR.Format(SR.net_WebSockets_AcceptUnsupportedProtocol, string.Join(", ", options.RequestedSubProtocols), subprotocol));
+                        }
+                        subprotocol = newSubprotocol;
+                    }
```

Comment:
Do we not need to save other headers for anything else?  Since we were reading here it seems like any data we don't save is just gone.